### PR TITLE
Move to React Native 0.70.1

### PR DESCRIPTION
**Pull request by wmathurin**
Successfully ran `SalesforceReact` tests (but one flaps sometimes).

### CODE REVIEW

File: iosTests/package.json [hunk 1]
```diff
@@ -10,9 +10,9 @@
     },
     "dependencies": {
         "create-react-class": "^15.7.0",
-        "react": "17.0.2",
-        "react-native": "0.67.1",
-        "react-native-force": "git+https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative.git#dev"
+        "react": "18.1.0",
+        "react-native": "0.70.1",
+        "react-native-force": "git+https://github.com/wmathurin/SalesforceMobileSDK-ReactNative.git#rn0701"
```
Comment (wmathurin) [inline iosTests/package.json:15]: Pointing to `wmathurin/SalesforceMobileSDK-ReactNative.git#rn0701` for now allow CI to succeed.
Will point to `forcedotcom/SalesforceMobileSDK-ReactNative.git#dev` before merging

File: iosTests/ios/Podfile [hunk 2]
```diff
@@ -13,15 +14,45 @@ end
...
+    :flipper_configuration => FlipperConfiguration.disabled,
+    :app_path => "#{Pod::Config.instance.installation_root}/.."
+  )
   pod 'React-RCTTest', :path => '../RCTTest'
   use_mobile_sdk!(:path => '../mobile_sdk/SalesforceMobileSDK-iOS')
-  pod 'SalesforceReact', :path => '../..'
+  pod 'SalesforceReact', :path => '../node_modules/react-native-force'
+end
+
+# To avoid Xcode 12 compilation errors in RNScreens and RNCMaskedView
```
Comment (brandonpage) [inline iosTests/ios/Podfile:33]: Xcode 12?  Should that be 13?

**Review by brandonpage — APPROVED** [review]

File: iosTests/ios/Podfile (hunk 2 above)
Comment (wmathurin) [reply iosTests/ios/Podfile:33]: It started in Xcode 12.

Comment (wmathurin) [reply iosTests/ios/Podfile:33]: That code is not new. I had it in the templates Podfile, I just made sure all the Podfile (tests / templates) are similar. We probably don't need it for the tests.

**Pull request merged**